# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [2]:
import pandas as pd
import numpy as np
import random

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000


In [35]:
first=pd.Series(np.random.randint(1,5,100))
second=pd.Series(np.random.randint(1,5,100))
third=pd.Series(np.random.randint(10000,30001,100))

### Step 3. Let's create a DataFrame by joinning the Series by column

In [36]:
df = pd.concat([first,second,third], axis=1)

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [37]:
df.columns=['bedrs', 'bathrs', 'price_sqr_meter']

    bedrs  bathrs  price_sqr_meter
0       4       3            24415
1       2       3            10208
2       4       2            21327
3       2       3            14929
4       2       4            29196
5       2       1            27643
6       3       1            22584
7       1       3            25521
8       4       2            15275
9       2       4            11020
10      3       3            27976
11      4       2            28108
12      1       1            15351
13      4       1            18693
14      4       4            13131
15      2       1            26063
16      4       4            23322
17      1       2            12813
18      4       4            15813
19      3       1            27119
20      2       4            16951
21      2       3            17038
22      2       2            26756
23      4       3            22059
24      2       2            29051
25      3       1            12132
26      4       4            26671
27      1       3   

### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [50]:
bigcolumn = pd.concat([first,second,third])
bigcolumn.reset_index(drop=True, inplace=True)
bigcolumn.tail

<bound method Series.tail of 0          4
1          2
2          4
3          2
4          2
5          2
6          3
7          1
8          4
9          2
10         3
11         4
12         1
13         4
14         4
15         2
16         4
17         1
18         4
19         3
20         2
21         2
22         2
23         4
24         2
25         3
26         4
27         1
28         3
29         2
       ...  
270    28287
271    27692
272    14095
273    12350
274    11811
275    10641
276    15593
277    27261
278    22066
279    26167
280    24876
281    18033
282    13390
283    14255
284    12049
285    28934
286    19361
287    18742
288    16503
289    26756
290    27490
291    10609
292    17906
293    10285
294    29779
295    29352
296    19831
297    17316
298    15166
299    11984
dtype: int64>

### Step 6. Ops it seems it is going only until index 99. Is it true?

In [43]:
bigcolumn.index

Int64Index([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9,
            ...
            90, 91, 92, 93, 94, 95, 96, 97, 98, 99],
           dtype='int64', length=300)

### Step 7. Reindex the DataFrame so it goes from 0 to 299

In [51]:
# bigcolumn.index = range(0,300)
bigcolumn.reset_index(drop=True,inplace=True)
bigcolumn.tail()

295    29352
296    19831
297    17316
298    15166
299    11984
dtype: int64